File: 7.reverse-integer.cpp
class Solution {                      
public:
    int reverse(int x) {
        long r=0;      // decleare r 
        while(x){
         r=r*10+x%10; // find reminder and add its to r
         x=x/10;     // Update the value of x
        }
        if(r>INT_MAX || r<INT_MIN) return 0; // check 32 bit range 
        return int(r);
    }
}; 

File: 7-reverse-integer/7-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        res = 0
        negative = (x < 0)

        if negative:
            x = -x

        while x:
            res = res*10
            res += (x%10)
            x = x//10

        if (res > 2147483648):
            return 0

        if negative:
            return -res

        return res